sql/sql_delete.cc

```diff
@@ -1182,25 +1237,46 @@ multi_delete::initialize_tables(JOIN *join)
       tbl->prepare_triggers_for_delete_stmt_or_event();
       tbl->prepare_for_position();
       tbl->file->prepare_for_modify(tbl->versioned(VERS_TIMESTAMP), true);
-    }
-    else if ((tab->type != JT_SYSTEM && tab->type != JT_CONST) &&
-             walk == delete_tables)
-    {
-      /*
-        We are not deleting from the table we are scanning. In this
-        case send_data() shouldn't delete any rows a we may touch
-        the rows in the deleted table many times
-      */
-      delete_while_scanning= false;
+
+      // maybe don't do this if join->need_tmp is false, dunno
```

Comment:
The code for new features should have more confidence :-)

Reply:
Removed incorrect comment.